src/USER-MEAMC/meam_setup_done.cpp

```diff
@@ -276,6 +282,21 @@ MEAM::compute_pair_meam(void)
               this->phir[nv2][j] = this->phir[nv2][j] - 0.75 * S11 * phiaa - 0.25 * S22 * phibb;
             }
 
+          } else if (this->lattce_meam[a][b] == DIA3 || this->lattce_meam[a][b] == LIN || 
+                    this->lattce_meam[a][b] == ZIG  || this->lattce_meam[a][b] == TRI) {
+            // To avoid nan values of phir due to rapid decrease of b2nn^n or/and 
+            // argument of phi_meam, i.e. r*arat^n, in some cases (3NN dia with low Cmin value)
+            if (scrn > 0.0) {            
+              b2nn = -Z2*scrn/Z1;
+              for (n = 1; n <= 10; n++) {
+                phi_val = MathSpecial::powint(b2nn,n)*phi_meam(r * MathSpecial::powint(arat, n), a, b);
+                if (!iszero(phi_val)){
+                  this->phir[nv2][j] += phi_val;
+                } else {
+                  break; // necessary to avoid numerical error (nan or infty) due to exponential decay in phi_meam 
+                }
+              }
+            }
```

Comment:
The more I look at this, the more I think we should just *always* do something like this, including the two aa/bb loops above. I'll refactor this into an function and use that... doesn't even have to be inlined, as we only come by here once.